crates/sui-framework/sources/governance/validator_set.move

```diff
@@ -42,6 +46,10 @@ module sui::validator_set {
 
         /// Mappings from staking pool's ID to the sui address of a validator.
         staking_pool_mappings: Table<ID, address>,
+
+        /// When a validator is deactivated the validator is removed, but its
+        /// staking pool remains here so that delegators can withdraw their stake from it.
+        inactive_pools: LinkedTable<ID, StakingPool>,
```

Comment:
This needs to be a `LinkedTable` so we can iterate through the inactive pools and destroy them in tests. Otherwise this can be a `Table`.

Reply:
Ah, interesting... see comment about about polymorphic destroy, I think this might be a good motivation to add this feature now, since I hate seeing prod code made unclear/less efficient to enable testing if it can be avoided.